### ISSUE TITLE
Model is not implemented ?

### ISSUE THREAD

**Issue by adesgautam** [naiveHobo/InvoiceNet#13]
Hi, I saw the invoicenet.commom.model module, it is not implemented yet ? 
When it will be implemented ?
And what kind of model is used here ?

**Comment by naiveHobo** [comment]
That's just an interface class that's used to implement the main ACP and Parser models.

The main ACP model is implemented [here](https://github.com/naiveHobo/InvoiceNet/blob/master/invoicenet/acp/acp.py). You can find the deeper model architecture and the parser models by following the imports from this module.

**Comment by adesgautam** [comment]
Okay thanks!
I just want to clarify about the process. 
The model uses pdf text to parse the required data right ? Or you can please clarify the process.

**Comment by naiveHobo** [comment]
Nope, PDF text is never used. I tested on a dataset of about 10000 documents and I finally decided to use an OCR in all cases. Using an OCR at the initial stage of data processing had some benefits:

- Enabled a universal pipeline for PDFs with embedded text, scanned documents, and images
- Some PDFs that had embedded text sometimes missed out on some important sections in the PDF (eg. the logo). Using the embedded text in this case meant there was no way to verify if I'm actually able to extract all readable significant text from the PDF or not.

Each input document is first converted to an image and the text along with the bounding box information is extracted using Tesseract. Using an OCR here means that the performance of the model will always be limited by the performance of the OCR, but I can say that Tesseract performed well for documents with a good resolution (which was true for about 95% of the dataset I used).
